public/webworker.js

```diff
@@ -101,13 +120,39 @@ async function startDatasette(settings) {
             db[bit].transform(
                 types=tracker.types
             )
+        for json_url in jsons:
+            bit = json_url.split("/")[-1].split(".")[0].split("?")[0]
+            bit = bit.strip()
+            if not bit:
+                bit = "table"
+            prefix = 0
+            base_bit = bit
+            while bit in table_names:
+                prefix += 1
+                bit = "{}_{}".format(base_bit, prefix)
+            table_names.add(bit)
+            response = await pyfetch(json_url)
+            with open("json.json", "wb") as fp:
+                json_bytes = await response.bytes()
+                try:
+                    json_data = json.loads(json_bytes)
+                except json.decoder.JSONDecodeError:
+                    # Maybe it's newline-delimited JSON?
+                    # This will raise an unhandled exception if not
+                    json_data = [json.loads(line) for line in json_bytes.splitlines()]
+            # If it's an object, try to find first key that's a list of objects
+            if isinstance(json_data, dict):
+                for key, value in json_data.items():
+                    if isinstance(value, list) and value and isinstance(value[0], dict):
+                        json_data = value
+            assert isinstance(json_data, list), "JSON data must be a list of objects"
+            db[bit].insert_all(json_data)
+
     from datasette.app import Datasette
     ds = Datasette(names, settings={
         "num_sql_threads": 0,
-    }, metadata = {
-        "about": "Datasette Lite",
-        "about_url": "https://github.com/simonw/datasette-lite"
-    })
+        "base_url": "${settings.baseUrl}"
```

Comment:
This was needed to support https://github.com/rclement/datasette-dashboards/pull/103